Content.Server/Players/JobWhitelist/JobWhitelistManager.cs

```diff
@@ -0,0 +1,119 @@
+﻿using System.Linq;
+using System.Threading;
+using System.Threading.Tasks;
+using Content.Server.Database;
+using Content.Shared.CCVar;
+using Content.Shared.Players.JobWhitelist;
+using Content.Shared.Roles;
+using Robust.Server.Player;
+using Robust.Shared.Configuration;
+using Robust.Shared.Network;
+using Robust.Shared.Player;
+using Robust.Shared.Prototypes;
+using Robust.Shared.Utility;
+
+namespace Content.Server.Players.JobWhitelist;
+
+public sealed class JobWhitelistManager : IPostInjectInit
+{
+    [Dependency] private readonly IConfigurationManager _config = default!;
+    [Dependency] private readonly IServerDbManager _db = default!;
+    [Dependency] private readonly INetManager _net = default!;
+    [Dependency] private readonly IPlayerManager _player = default!;
+    [Dependency] private readonly IPrototypeManager _prototypes = default!;
+    [Dependency] private readonly UserDbDataManager _userDb = default!;
+
+    private readonly Dictionary<Guid, HashSet<string>> _whitelists = new();
+
+    public void Initialize()
+    {
+        _net.RegisterNetMessage<MsgJobWhitelist>();
+    }
+
+    private async Task LoadData(ICommonSession session, CancellationToken cancel)
+    {
+        var whitelists = await _db.GetJobWhitelists(session.UserId);
+        _whitelists[session.UserId] = whitelists.ToHashSet();
+    }
+
+    private void FinishLoad(ICommonSession session)
+    {
+        SendJobWhitelist(session);
+    }
+
+    private void ClientDisconnected(ICommonSession session)
+    {
+        _whitelists.Remove(session.UserId);
+    }
+
+    public async void AddWhitelist(Guid player, ProtoId<JobPrototype> job)
+    {
+        _whitelists.GetOrNew(player).Add(job);
+
+        try
+        {
+            await _db.AddJobWhitelist(player, job);
+        }
+        catch
+        {
+            // ignored
+        }
+
+        if (_player.TryGetSessionById(new NetUserId(player), out var session))
+            SendJobWhitelist(session);
+    }
+
+    public bool IsAllowed(Guid player, ProtoId<JobPrototype> job)
```

Comment:
Fixed in [`79d00c6` (#28085)](https://github.com/space-wizards/space-station-14/pull/28085/commits/79d00c6c52d52f82c375726208b46ef0f2f95c1c)
This adds a try get session from id call in StationJobsGetCandidatesEvent, since the method raising that gets the NetUserId of the players instead of the ICommonSession.